### ISSUE TITLE
Text Field Widget

### ISSUE THREAD

**Issue by DraftedDev** [maycoon-ui/maycoon#12]

**Comment by waywardmonkeys** [comment]
We're working towards a new release of Parley in the Linebender community which contains some stuff to make this easier based on what was done in Masonry (and Druid before that).

You might find that useful. Parley builds on the Google font stack (skrifa) which is already in your dependency tree due to Vello, so it might not be too bad to switch from `font-kit` to `parley` and `fontique` ...

**Comment by DraftedDev** [comment]
> We're working towards a new release of Parley in the Linebender community which contains some stuff to make this easier based on what was done in Masonry (and Druid before that).
> 
> You might find that useful. Parley builds on the Google font stack (skrifa) which is already in your dependency tree due to Vello, so it might not be too bad to switch from `font-kit` to `parley` and `fontique` ...

Thanks for bringing this to my attention. I appreciate every help.

We already have a "Text" widget to display text. This issue is about a text input field. Sorry for the confusion.

**Comment by waywardmonkeys** [comment]
Right, I understand. We haven't had a recent enough Parley release yet, but in the next version, there is an `Editor` which is for helping to implement input fields and text area inputs:

https://github.com/linebender/parley/blob/c8e26f0d9f0812e41cb790f3d769521343a0f0cb/parley/src/layout/editor.rs#L100

This is used with `masonry` in the editable text area: https://github.com/linebender/xilem/blob/d77b31b892cfc5390c3fd49bc2f41da4101e3f24/masonry/src/widgets/text_area.rs#L136

It is for helping out with cursor management and other things that are needed in a text box.

If you want to drop by the Linebender Zulip (http://xi.zulipchat.com) sometime to chat with us (and especially the more text-oriented people), feel free!
